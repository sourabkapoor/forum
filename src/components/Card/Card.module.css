.card {
  background-color: white;
  color: var(--secondary-color);
  padding: 12px;
  font-size: 14px;
  /* border: 1px solid #474748; */
  box-shadow: rgba(0, 0, 0, 0.24) 0px 3px 8px;
  border-radius: 20px;
  overflow: visible;
  word-wrap: break-word;
  min-width: 312px;
  margin-bottom: 16px;
}

@media (max-width: 640px) {
  .card {
    min-width: unset;
    max-width: 94%;
  }
}
