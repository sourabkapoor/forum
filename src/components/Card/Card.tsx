import CSSModules from "react-css-modules";
import styles from "./Card.module.css";

interface Props {
  children: JSX.Element | JSX.Element[];
}

const Card: React.FC<Props> = ({ children }) => {
  return <div styleName="card">{children}</div>;
};

export default CSSModules(Card, styles, {
  allowMultiple: true,
  handleNotFoundStyleName: "log",
});
