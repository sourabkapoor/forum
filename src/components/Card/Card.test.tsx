import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import Card from "./Card";

describe("Card", () => {
  test("snapshot", () => {
    const { asFragment } = render(
      <Card>
        <div>Mock Children</div>
      </Card>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
