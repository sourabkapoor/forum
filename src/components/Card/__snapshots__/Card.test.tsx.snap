// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card snapshot 1`] = `
<DocumentFragment>
  <div
    class="card"
  >
    <div>
      Mock Children
    </div>
  </div>
</DocumentFragment>
`;
